src/cmds/openapi/reduce.ts

```diff
@@ -0,0 +1,189 @@
+import type { CommandOptions } from '../../lib/baseCommand';
+
+import fs from 'fs';
+import path from 'path';
+
+import chalk from 'chalk';
+import jsonpath from 'jsonpath';
+import oasReducer from 'oas/dist/lib/reducer';
+import ora from 'ora';
+
+import Command, { CommandCategories } from '../../lib/baseCommand';
+import { oraOptions } from '../../lib/logger';
+import prepareOas from '../../lib/prepareOas';
+import promptTerminal from '../../lib/promptWrapper';
+
+export type Options = {
+  spec?: string;
+  workingDirectory?: string;
+};
+
+export default class OpenAPIReduceCommand extends Command {
+  constructor() {
+    super();
+
+    this.command = 'openapi:reduce';
+    this.usage = 'openapi:reduce [file] [options]';
+    this.description = 'Reduce an OpenAPI definition into a smaller subset.';
+    this.cmdCategory = CommandCategories.APIS;
+    this.position = 3;
+
+    this.hiddenArgs = ['spec'];
+    this.args = [
+      {
+        name: 'spec',
+        type: String,
+        defaultOption: true,
+      },
+      {
+        name: 'workingDirectory',
+        type: String,
+        description: 'Working directory (for usage with relative external references)',
+      },
+    ];
+  }
+
+  async run(opts: CommandOptions<Options>) {
+    super.run(opts);
+
+    const { spec, workingDirectory } = opts;
+
+    if (workingDirectory) {
+      process.chdir(workingDirectory);
+    }
+
+    const { bundledSpec, specPath, specType } = await prepareOas(spec, 'openapi:reduce');
+    const parsedBundledSpec = JSON.parse(bundledSpec);
+
+    if (specType !== 'OpenAPI') {
+      throw new Error('Sorry, this reducer feature in rdme only supports OpenAPI 3.0+ definitions.');
+    }
+
+    const promptResults = await promptTerminal([
+      {
+        type: 'select',
+        name: 'reduceBy',
+        message: 'Would you like to reduce by paths or tags?',
+        choices: [
+          { title: 'Tags', value: 'tags' },
+          { title: 'Paths', value: 'paths' },
+        ],
+      },
+      {
+        type: (prev, values) => (values.reduceBy === 'tags' ? 'multiselect' : null),
+        name: 'tags',
+        message: 'Choose which tags to reduce by:',
+        min: 1,
+        choices: () => {
+          const tags = jsonpath
+            .query(parsedBundledSpec, '$..paths..tags')
+            .flat()
+            .filter(tag => {
+              // A spec may have a `name: tag` which this JSONPath query will pick up. Since that
+              // definitely isn't a tag we want here we need to filter them out.
+              return typeof tag === 'string';
```

Comment:
damn nice find... I wish there was an easy way to test this but can't really think of one 🤔

Reply:
Yeah I'm not sure how to test what a prompt in `prompts` is rendering without factoring this work out into a separate small utility.